### PR TITLE
support computing proof with n arguments

### PR DESCRIPTION
**Pull request by atoulme**
Signed-off-by: Antoine Toulme <antoine@lunar-ocean.com>

Instead of using 4 arguments, allow to use n arguments for the commit method.

### CODE REVIEW

**Review by shemnon — APPROVED** [review]

File: ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java [hunk 1]
```diff
@@ -19,14 +19,29 @@
...
-    byte[] result = LibIpaMultipoint.commit(input.toArrayUnsafe(), input.toArrayUnsafe(), input.toArrayUnsafe(), input.toArrayUnsafe());
+    byte[] result = LibIpaMultipoint.commit(new byte[][]{input.toArrayUnsafe(), input.toArrayUnsafe(), input.toArrayUnsafe(), input.toArrayUnsafe()});
     assertThat(Bytes.wrap(result)).isEqualTo(Bytes.fromHexString("0xc70a1e0077e1fff6702f2bde0cccf1bf5915d4c5ab73c33ea5e75b6f703d2346aa7aa373cd07fdf684282c11a7f7623b6e67d2b65862ca0011e2415726c87415d07ada26afff5e6be8066c57228e78399cb3af7490f4de739eef0191907eca0d9ba47aec3457f0fab28324061ec27508e29b067acb3a97fbb43dea61a376f73b"));
   }
+
+  @Test
+  public void testCallLibraryWithManyElements() {
+    Bytes input = Bytes.fromHexString("0x0cfe3041fb6512c87922e2146c8308b372f3bf967f889e69ad116ce7c7ec840cfe3041fb6512c87922e2146c8308b372f3bf967f889e69ad116ce7c7ec840cfe3041fb6512c87922e2146c8308b372f3bf967f889e69ad116ce7c7ec840cfe3041fb6512c87922e2146c8308b372f3bf967f889e69ad116ce7c7ec84");
+    List<byte[]> params = new ArrayList<>();
+    for (int i = 0 ; i < 128; i++) {
```
Comment (dragan2234) [inline ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java:41]: hey @atoulme i've been looking into the ipa-multipoint wrapper, and have one question. Where are these numbers pulled from? I mean `0x0cfe3041fb6512c87922e2146c8308b372f3bf967f889e69ad116ce7c7ec840cfe3041fb6512c87922e2146c8308b372f3bf967f889e69ad116ce7c7ec840cfe3041fb6512c87922e2146c8308b372f3bf967f889e69ad116ce7c7ec840cfe3041fb6512c87922e2146c8308b372f3bf967f889e69ad116ce7c7ec84` for example

Also, should we make the for loop go up to 256 values?

Comment (atoulme) [reply ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java:41]: It’s a test. I picked up this number, ran it with go impl, then used it here to create a test.

Comment (dragan2234) [reply ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java:41]: @atoulme thanks, makes sense. Btw did you mean you ran it with rust impl?

Comment (atoulme) [reply ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/LibIpaMultipointTest.java:41]: No with the go impl. Though rust should match

**Pull request merged**